meta-calculinux-apps/recipes-apps/upsilon/upsilon_git.bb

```diff
@@ -0,0 +1,114 @@
+SUMMARY = "Upsilon graphing calculator firmware for Picocalc"
+DESCRIPTION = "Upsilon is a fork of Omega, a user-made OS for NumWorks calculator, \
+               adapted for the Picocalc device with enhanced features including \
+               Kandinsky module, wallpapers, external apps, and Python improvements."
+HOMEPAGE = "https://github.com/gurubook/Upsilon"
+LICENSE = "CC-BY-NC-SA-4.0"
+LIC_FILES_CHKSUM = "file://LICENSE.md;md5=91bec4f3756d037e29191d67b4737e63"
+
+# Using upsilon-dev branch for now
+# TODO: Switch to a stable release tag once available (e.g., SRCREV = "${AUTOREV}" and PV = "1.0+git${SRCPV}")
+# Current releases at https://github.com/gurubook/Upsilon/releases are minimal
+# Note: submodules=1 should fetch rpn and atomic apps, but may need manual initialization
+PV = "1.0+git${SRCPV}"
+SRC_URI = "git://github.com/gurubook/Upsilon.git;protocol=https;branch=upsilon-dev;submodules=1"
+SRCREV = "${AUTOREV}"
+
+# Pin to specific commit for reproducibility in production:
+# SRCREV = "<commit-hash>"
+
+S = "${WORKDIR}/git"
+
+DEPENDS = " \
+    libsdl2 \
+    libpng \
+    jpeg \
+    freetype \
+    freetype-native \
+    libdrm \
+    virtual/libgles2 \
+    python3-native \
+    imagemagick-native \
+    pkgconfig-native \
+"
+
+# Only compatible with Picocalc target
+COMPATIBLE_MACHINE = "luckfox-lyra"
```

Comment:
this doesn't need to be here. this should be compatible with other arches as well once we have any..